openhcl/underhill_core/src/nvme_manager.rs

```diff
@@ -159,17 +160,20 @@ impl NvmeManagerClient {
 }
 
 #[derive(Inspect)]
-struct NvmeManagerWorker {
+struct NvmeManagerWorker<F> {
     #[inspect(skip)]
     driver_source: VmTaskDriverSource,
     #[inspect(iter_by_key)]
     devices: HashMap<String, nvme_driver::NvmeDriver<VfioDevice>>,
     #[inspect(skip)]
-    dma_buffer: Arc<dyn VfioDmaBuffer>,
+    dma_buffer_spawner: F,
```

Comment:
I'd make this a `Box<dyn FnMut(...)>` rather than add a generic parameter.

Reply:
neat, will swap